app/assets/javascripts/autoform_form.js

```diff
@@ -19,14 +20,57 @@ $(document).ready(function() {
             return false;
         }
         else {
-            $('#autoform-error').hide();
-            $('#autoform-email').parent('.form-group').removeClass('has-error');
-            $('#autoform-domain').parent('.form-group').removeClass('has-error');
-            $('#autoform-form').hide();
-            $('#autoform-form-sent').show();
+            const formData = $(this).serialize();
+
+            $.post($(this).attr('action'), formData)
+                .done(function() {
+                    $('#autoform-error').hide();
+                    $('#autoform-form-error').hide();
+                    $('#autoform-email').parent('.form-group').removeClass('has-error');
+                    $('#autoform-domain').parent('.form-group').removeClass('has-error');
+                    $('#autoform-form').hide();
+                    $('#autoform-form-sent').show();
+                })
+                .fail(function() {
+                    $('#autoform-form-error').show();
+                });
 
             plausible('Registration', {props: {product: 'Autoform'}});
             return true;
         }
     });
+
+    const autoformError = document.getElementById('autoform-error');
+    addCloseButtonListener(autoformError);
+
+    const autoformFormError = document.getElementById('autoform-form-error');
+    addCloseButtonListener(autoformFormError);
+
+    const emailInputField = document.getElementById('autoform-email');
+    emailInputField.addEventListener('click', function (){
+        removeErrors(autoformError, autoformFormError)
+    });
+
+    const domainInputField = document.getElementById('autoform-domain');
+    domainInputField.addEventListener('click', function (){
+        removeErrors(autoformError, autoformFormError)
+    });
 });
+
+function addCloseButtonListener(node) {
+    const closeButton = node.querySelector('.alert .close');
+    closeButton.addEventListener("click", function (){
+        removeErrors(node);
+    });
+}
+
+function removeErrors(...nodes) {
```

Comment:
tu tiež tie 3 bodky musia ísť preč, nepozná to takú syntax. Ale malo by to fungovať úplne rovnako aj bez tých bodiek.